src/wing/tigletaxsifunctions.cpp

```diff
@@ -225,68 +225,114 @@ void InterpolateXsi(const std::string& refUID1, const EtaXsi& etaXsi1,
         throw CTiglError("Eta not in range [0,1] in InterpolateXsi.", TIGL_MATH_ERROR);
     }
 
-    // make sure all uids refer to the same wing
-    const auto& wingcs1 = _getBelongingComponentSegment(refUID1, uidMgr);
-    const auto& wingcs2 = _getBelongingComponentSegment(refUID2, uidMgr);
-    const auto& wingcs3 = _getBelongingComponentSegment(targetUID, uidMgr);
+    // declare points to compute the (component) segment line
+    gp_Pnt p1;
+    gp_Pnt p2;
+
+    // declare points to compute the iso eta line of segment
+    gp_Pnt pLE;
+    gp_Pnt pTE;
+
+    // check if all three UIDs in the argument refer to the same segment
+    const CTiglUIDManager::TypedPtr tp1 = uidMgr.ResolveObject(refUID1);
+    const CTiglUIDManager::TypedPtr tp2 = uidMgr.ResolveObject(refUID2);
+    const CTiglUIDManager::TypedPtr tpTarget = uidMgr.ResolveObject(targetUID);
```

Comment:
I think we should remove the `TypedPtr` from the public interface of the uid manager. Relying on `reinterpret_cast` is code smell and casting to and from `void*` is potentially dangerous. We already have an [any](https://github.com/DLR-SC/tigl/blob/master/src/common/any.h) implementation we could use instead, and we could switch to `std::any` once we allow C++17. But this is something for another issue. 

In the meantime, here, I suggest to use the templated `CTiglUIDManager::IsType` method and the templated `CTiglUIDManager::ResolveObject` methods to check for the type and resolve to an actual type here. These functions at least don't `reinterpret_cast` under the hood and we can hide ugly type casting under the hood of a clean API.

I realize that this is not part of your change, but let's use this as a chance for some clean up :)

Reply:
Thanks for this remark! I implemented your suggested solution with `IsType` and `ResolveObject`. The outlook of using std::any once we update to C++ 17 sounds interesting!